src/pykeen/evaluation/rank_based_evaluator.py

```diff
@@ -27,19 +29,60 @@
 SIDE_HEAD = 'head'
 SIDE_TAIL = 'tail'
 SIDE_BOTH = 'both'
+SIDES = {SIDE_HEAD, SIDE_TAIL, SIDE_BOTH}
+
 RANK_OPTIMISTIC = 'optimistic'
 RANK_PESSIMISTIC = 'pessimistic'
 RANK_REALISTIC = 'realistic'
-RANK_EXPECTED_REALISTIC = 'expected_realistic'
 RANK_TYPES = {RANK_OPTIMISTIC, RANK_PESSIMISTIC, RANK_REALISTIC}
-SIDES = {SIDE_HEAD, SIDE_TAIL, SIDE_BOTH}
 
-MEAN_RANK = 'mean_rank'
-MEAN_RECIPROCAL_RANK = 'mean_reciprocal_rank'
-ADJUSTED_MEAN_RANK = 'adjusted_mean_rank'
-ADJUSTED_MEAN_RANK_INDEX = 'adjusted_mean_rank_index'
-TYPES_ALL = {MEAN_RANK, MEAN_RECIPROCAL_RANK}
-TYPES_REALISTIC_ONLY = {ADJUSTED_MEAN_RANK, ADJUSTED_MEAN_RANK_INDEX}
+RANK_EXPECTED_REALISTIC = 'expected_realistic'
+EXPECTED_RANKS = {
+    RANK_REALISTIC: RANK_EXPECTED_REALISTIC,
+    RANK_OPTIMISTIC: None,  # TODO - research problem
+    RANK_PESSIMISTIC: None,  # TODO - research problem
+}
+
+ARITHMETIC_MEAN_RANK = 'arithmetic_mean_rank'  # also known as mean rank (MR)
```

Comment:
is changing this going to be a problem?

Reply:
I guess so, since mean_rank might have been used before. Maybe we can add synonyms for those, similar to what we have e.g. with losses?

Reply:
I just worked on the automatic canonicalization of the names used, so anyone who uses something like "mean_rank" in their code for HPO or interaction with the results class will be okay.

If they're hard-coding access to the dictionary after export, they'll just have to upgrade. I think it's worth it to make things more descriptive. Counterpoint - this is me being principled over practical.